src/main/java/com/marginallyclever/makelangelo/firmwareUploader/FirmwareUploaderPanel.java

```diff
@@ -74,7 +74,7 @@ private void updateCOMPortList() {
 	}
 	
 	private String[] getListOfCOMPorts() {
-		return SerialTransportLayer.listConnections();
+		return new SerialTransportLayer().listConnections().toArray(new String[0]);
```

Comment:
Why is it better to have SerialTransportLayer *not* be a singleton?

Reply:
It was not a real singleton, it was a static method but you can say it is the same thing :)
In the whole application, there is few class containing only static method which is a good thing. It will be useful when new unit test will be added, it will be easier to mock it.